module/experiment.js

```diff
@@ -2,27 +2,46 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
 import { Variant } from './variant'
-import { setExperimentVariant } from './actions'
+import { resetExperimentVariant } from './actions'
 
 export function Selector (props) {
-	const { variant, name, children } = props
-	let chosenOne
-	React.Children.forEach(children, child => {
-		if (child.type === Variant && child.props.name === variant) {
-			chosenOne = React.cloneElement(child, { experiment: name })
-		}
-	})
+    const { variant, name, children, resetExperimentVariant } = props
+    const variants = getComponentVariants(children)
+    let chosenOne
+    if (variant === null) {
+        resetExperimentVariant(name, variants)
```

Comment:
I don't think you you should dispatch from the render function. Better check for an invalid variant in the setExperimentVariant codepath